test/jasmine/tests/cartesian_interact_test.js

```diff
@@ -2547,3 +2547,123 @@ describe('Cartesian plots with css transforms', function() {
         });
     });
 });
+
+describe('Cartesian taces with zindex', function() {
+    var gd;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    var data0 = [
+        {x: [1, 2], y: [1, 1], type: 'scatter', zindex: 10},
+        {x: [1, 2], y: [1, 2], type: 'scatter'},
+        {x: [1, 2], y: [1, 3], type: 'scatter', zindex: 5}
+    ];
+
+    var data1 = [
+        {x: [1, 2], y: [1, 1], type: 'scatter'},
+        {x: [1, 2], y: [1, 2], type: 'scatter', zindex: -5},
+        {x: [1, 2], y: [1, 3], type: 'scatter', zindex: 10},
+    ];
```

Comment:
Lets' adjust `marker.sizes` to make it easier to see what's going on when/if one wants to debug this.
```suggestion
    var data0 = [
        {x: [1, 2], y: [1, 1], type: 'scatter', marker: {size: 10}, zindex: 10},
        {x: [1, 2], y: [1, 2], type: 'scatter', marker: {size: 30}},
        {x: [1, 2], y: [1, 3], type: 'scatter', marker: {size: 20}, zindex: 5}
    ];

    var data1 = [
        {x: [1, 2], y: [1, 1], type: 'scatter', marker: {size: 10}},
        {x: [1, 2], y: [1, 2], type: 'scatter', marker: {size: 30}, zindex: -5},
        {x: [1, 2], y: [1, 3], type: 'scatter', marker: {size: 20}, zindex: 10},
    ];
```